src/printer_internal.h

```diff
@@ -122,6 +122,18 @@ LY_ERR yin_print_parsed_module(struct ly_out *out, const struct lys_module *modu
 LY_ERR yin_print_parsed_submodule(struct ly_out *out, const struct lys_module *module,
         const struct lysp_submodule *submodp, uint32_t options);
 
+/**
+ * @brief YANG Tree Diagram printer of the parsed submodule. Full YANG Tree printer.
+ *
+ * @param[in] out Output specification.
+ * @param[in] module Main module.
+ * @param[in] options Schema output options (see @ref schemaprinterflags).
+ * @param[in] line_length Maximum characters to be printed on a line, 0 for unlimited. Only for ::LYS_OUT_TREE printer.
+ * @return LY_ERR value, number of the printed bytes is updated in ::ly_out.printed.
+ */
+LY_ERR tree_print_parsed_module(struct ly_out *out, const struct lys_module *module, uint32_t options,
```

Comment:
I would argue that the compiled module is printed more than the parsed module. Maybe omit `parsed`?

Reply:
I think that right now it prints information from the parsed tree. It is prepared to add also the compiled tree, but the format specification requires things like augments or groupings. So, the decision was to use the parsed trees for now in cost of for example not expanding uses (which is allowed by format specs)

Reply:
Right, makes sense then.